src/frontend/hooks/useProjectInvitesListener.ts

```diff
@@ -0,0 +1,58 @@
+import {useCallback, useEffect, useState} from 'react';
+import {useApi} from '../contexts/ApiContext';
+import {useQueryClient} from '@tanstack/react-query';
+import {INVITE_KEY} from './server/invites';
+import {MapBuffers} from '@mapeo/core/dist/types';
+import {InviteInternal, InviteRemovalReason} from '@mapeo/core/dist/invite-api';
+
+export const useProjectInvitesListener = ({
```

Comment:
i think this hook is doing too much and the bottom sheet-specific pieces can be extracted to a separate hook.

for example, i would basically restore `initializeInviteListener()`, but without the piece related to the `invite-removed` event:

```ts
export const useProjectInvitesListener = () => {
  const mapeoApi = useApi();
  const queryClient = useQueryClient();

  useEffect(() => {
    function resetInvitesCache() {
      queryClient.invalidateQueries({queryKey: [INVITE_KEY]});
    }

    mapeoApi.invite.addListener('invite-received', resetInvitesCache);

    return () => {
      mapeoApi.invite.removeListener('invite-received', resetInvitesCache);
    };
  }, [queryClient, mapeoApi]);
};
```

then define a separate hook for the `invite-removed` event:

```ts
function useCancelledInvite(
  inviteId: string,
  bottomSheetIsOpen: boolean,
) {
  const mapeoApi = useApi();
  const queryClient = useQueryClient();

  const [cancelled, setCancelled] = useState(false);

  const dismissCancelledInvite = useCallback(() => {
    setCancelled(false);
    queryClient.invalidateQueries({queryKey: [INVITE_KEY]});
  }, [queryClient, setCancelled]);

  useEffect(() => {
    function shouldInterceptCancel(
      val: MapBuffers<InviteInternal>,
      reason: InviteRemovalReason,
    ) {
      if (
        reason === 'canceled' &&
        inviteId === val.inviteId &&
        bottomSheetIsOpen
      ) {
        setCancelled(true);
      }
    }

    mapeoApi.invite.addListener('invite-removed', shouldInterceptCancel);

    return () => {
      mapeoApi.invite.removeListener('invite-removed', shouldInterceptCancel);
    };
  }, [mapeoApi, inviteId, bottomSheetIsOpen]);

  return [cancelled, dismissCancelled] as const;
}
```

`useProjectInvitesListener()` could then be used anywhere near the top level of the app, similar to what `initializeInviteListener()` was doing. `useCancelledInvite()` would then be used in `ProjectInviteBottomSheet`

Reply:
thinking about this more, and maybe the separation is more tricky than i thought. i think that's a symptom of how the invites are set up and i think the potential refactor i'm thinking of would fix it.

if it turns out that this separation that i'm proposing isn't feasible, can potentially ignore it for now.